lib/daemon.py

```diff
@@ -4,47 +4,22 @@
 
 from systemd.daemon import notify  # type: ignore
 
-from uaclient import defaults, http
+from uaclient import http
 from uaclient.config import UAConfig
-from uaclient.daemon import (
-    poll_for_pro_license,
-    retry_auto_attach,
-    setup_logging,
-)
+from uaclient.daemon import poll_for_pro_license, retry_auto_attach
+from uaclient.log import setup_journald_logging
 
-LOG = logging.getLogger("ubuntupro.lib.daemon")
+LOG = logging.getLogger("ubuntupro.daemon")
 
 
 def main() -> int:
-    setup_logging(
-        logging.INFO,
-        logging.DEBUG,
-        defaults.CONFIG_DEFAULTS["daemon_log_file"],
-        logger=LOG,
-    )
-    cfg = UAConfig()
-    setup_logging(
-        logging.INFO, logging.DEBUG, log_file=cfg.daemon_log_file, logger=LOG
-    )
-    # used with loggers in uaclient.daemon
-    daemon_logger = logging.getLogger("ubuntupro.daemon")
-    setup_logging(
-        logging.INFO,
-        logging.DEBUG,
-        log_file=cfg.daemon_log_file,
-        logger=daemon_logger,
-    )
-    # The ua-daemon logger should log everything to its file
-    # Make sure the ua-daemon logger does not generate double logging
-    # by propagating to the root logger
+    setup_journald_logging(logging.DEBUG, LOG)
     LOG.propagate = False
-    daemon_logger.propagate = False
-    # The root logger should only log errors to the daemon log file
-    setup_logging(
-        logging.CRITICAL,
-        logging.ERROR,
-        cfg.daemon_log_file,
-    )
+    setup_journald_logging(logging.ERROR, logging.getLogger("ubuntupro"))
+
+    cfg = UAConfig()
+
+    # Make sure the ubuntupro.daemon logger does not generate double logging
```

Comment:
This comment should be above the LOG.propagate line

Reply:
Ah, forgot this.